fix serveral issues and add more corrupted images

jaysim opened the pull request.

jaysim committed:
ntfsck: add new corrupted images

NEED_TO_FIX/lcn_bitmap_clear_remain/ntfs.img.tar.xz
 after fsck, still remained lcn bitmap to be cleared
NEED_TO_FIX/memory_leak1/ntfs.img.tar.xz
 many memory leak occurred
NEED_TO_FIX/mftmirror_file_load_failure/ntfs.img.tar.xz
 $MFTMirror load failure image
remove_index_entry_twice/ntfs.img.tar.xz
 remove same index entry twice, so error remained (false positive)

Signed-off-by: JaeHoon Sim <jay.sim@lge.com>

jaysim committed:
ntfsck: prevent orphaned inode with different sequence number frome being added

Missing comparision with parents' sequence number and orphaned file's sequence
number in adding orphaned inode.
Add checking both sequence number.

simplify(?) 'goto' label.

Signed-off-by: JaeHoon Sim <jay.sim@lge.com>

namjaejeon commented on src/ntfsck.c:
```diff
@@ -878,6 +878,56 @@ static int ntfsck_add_inode_to_lostfound(ntfs_inode *ni, FILE_NAME_ATTR *fn,
+	ntfsck_mft_bmp_bit_clear(mft_no);
+}
+
+static void __ntfsck_delete_orphaned_inode(ntfs_inode **ni)
```
any reason why __ is added as prefix ?

namjaejeon commented on src/ntfsck.c:
```diff
@@ -878,6 +878,56 @@ static int ntfsck_add_inode_to_lostfound(ntfs_inode *ni, FILE_NAME_ATTR *fn,
 }
 
+/* delete orphaned mft, call this when inode open failed. */
+static void __ntfsck_delete_orphaned_mft(ntfs_volume *vol, u64 mft_no)
```
__ prefix? why?

jaysim replied:
I removed it !

namjaejeon commented on src/ntfsck.c:
```diff
@@ -989,6 +1036,12 @@ static int ntfsck_add_index_entry_orphaned_file(ntfs_volume *vol, s64 mft_no)
 			parent_ni = ntfs_inode_open(vol, MREF(parent_no));
 
 		if (parent_ni) {
+			if (__ntfsck_cmp_parent_mft_sequence(parent_ni, fn)) {
```
__ prefix ? Why?

namjaejeon commented on src/ntfsck.c:
```diff
@@ -989,6 +1036,12 @@ static int ntfsck_add_index_entry_orphaned_file(ntfs_volume *vol, s64 mft_no)
 		if (parent_ni) {
+			if (__ntfsck_cmp_parent_mft_sequence(parent_ni, fn)) {
+				/* do not add inode to parent */
+				ntfs_log_info("##### Different seqnence number\n");
```
Probably we can add two sequence numbers in this print.

jaysim committed:
ntfsck: add checking parent mft number/sequence before check inode

There are some cases that parent mft number and sequence is different
with those of inode to be checked.
Add checking parent mft number and sequence.

Signed-off-by: JaeHoon Sim <jay.sim@lge.com>

jaysim committed:
ntfsck: new corrupted image of different parent mft number and sequence

tests/different_index_and_mft_flags/ntfs.img.tar.xz
  index inode and parent of inode is different

Signed-off-by: JaeHoon Sim <jay.sim@lge.com>
